### DISCUSSION TITLE
newbie help with getStaticProps

### DISCUSSION THREAD

**Discussion by jonsaul** [vercel/next.js#31118] [Help] [answered]
Hi all,

Was hoping someone might be able to give me some advice on an issue I have with my first next.js site build.

I have a collection of mdx files in a directory "/projects" ...

In my index.js I use getStaticProps to return a projects object, which I then pass into a component `<ProjectList projects={projects}  />` which then maps over the projects and  outputs a link to each project via a [slug].js template.

The issue I'm having is I would like to be able to output the full set of projects again in the same fashion as in my index.js from within my [slug].js, which is already using getStaticProps to provide data to the single (slug) template. I'm having some difficulty working out the best way to integrate these two tasks, and essentially also map out all my projects from within the [slug].js template as I do in index.js...

Hopefully showing some code is more clear than I can explain here... 

Index.js:

```javascript

import fs from 'fs'
import path from 'path'
import matter from 'gray-matter'

import ProjectList from '../components/project-list'
import IndexTitle from '../components/index-title'
import Footer from '../components/footer'


const index = ({ projects }) => (
    <>
      <IndexTitle />
      <ProjectList projects={projects.filter((project) => project.frontMatter.featured === true)} />
      <Footer />
    </>
);




const getStaticProps = async () => {
  const files = fs.readdirSync(path.join('projects'))
  const projects = files.map(filename => {
    const markdownWithMeta = fs.readFileSync(path.join('projects', filename), 'utf-8')
    const { data: frontMatter } = matter(markdownWithMeta)
    return {
      frontMatter,
      slug: filename.split('.')[0]
    }
  })
  return {
    props: {
      projects
    }
  }
}

export {getStaticProps};
export default index;
```

[slug].js

```javascript
import { serialize } from 'next-mdx-remote/serialize'
import { MDXRemote } from 'next-mdx-remote'
import fs from 'fs'
import path from 'path'
import matter from 'gray-matter'
import ProjectText from '../../components/project-text'
import ImageFull from '../../components/image-full'
import ProjectMeta from '../../components/project-meta'
import Footer from '../../components/footer'

const components = { ProjectText, ImageFull, ProjectMeta  }

const ProjectPage = ({ frontMatter, mdxSource }) => {
  return (
      <div>  
          <ImageFull url={frontMatter.heroUrl} priority={true} />
          <MDXRemote {...mdxSource} components={components} />
          <ProjectMeta frontMatter={frontMatter} />
          <Footer />
      </div>
  )
}

const getStaticPaths = async () => {
  const files = fs.readdirSync(path.join('projects'))

  const paths = files.map(filename => ({
    params: {
      slug: filename.replace('.mdx', '')
    }
  }))

  return {
    paths,
    fallback: false
  }
}

const getStaticProps = async ({ params: { slug } }) => {
  const markdownWithMeta = fs.readFileSync(path.join('projects',slug + '.mdx'), 'utf-8')

  const { data: frontMatter, content } = matter(markdownWithMeta)
  const mdxSource = await serialize(content)

  return {
    props: {
      frontMatter,
      slug,
      mdxSource
    }
  }
}

export { getStaticProps, getStaticPaths }
export default ProjectPage
```

So ideally I'm looking for how I can achieve these two tasks using the same getStaticProps on my [slug].js template and then include my `<ProjectList />` component to display all the projects within [slug].js

Any advice greatly appreciated.

Thanks!

**Comment by jeffcad** [comment]
Do you mean that you're trying to use the results of 1 getStaticProps (gSP) in 2 different pages? If yes, I don't think you can do that. I'd repeat the code in both gSP (or make a shared function in a lib folder and import into both gSP). If you think about how gSP works, it has to do its job even if somebody skips the index (home) page and goes straight to the slug with a bookmark or search result. If the gSP was only in 1 file, it wouldn't trigger.

**Reply by jonsaul** [reply]
No, I'm not looking for for one gSP to provide 2 different pages, I am struggling to get the results of the props for the single post, and also the results for the full projects data available to use within my slug template, I understand I need to repeat the code (almost merge the two gSP) im just struggling with the implementation on that, seems to result in conflicts each time I try to pull in all the projects data with the single project data in the same gSP in the slug template, sorry I wasn't sure if the post was very clear or not...

**Comment by jonsaul** [comment, marked as answer]
OK, I have now worked out that the issue I was experiencing was not actually related to getStaticProps at all, but was in fact a routing issue. I would delete the above post, but don't believe I can.
